[DataGrid] Column spanning (#4020)
mui/mui-x: docs/data/data-grid/columns/columns.md

{{"demo": "ColumnPinningApiNoSnap.js", "bg": "inline", "hideToolbar": true}}

## Column spanning

By default, each cell takes up the width of one column.
You can modify this behavior with column spanning.
It allows cells to span multiple columns.
This is very close to the "column spanning" in an HTML `<table>`.

To change the number of columns a cell should span, use the `colSpan` property available in `GridColDef`:

```ts
interface GridColDef {
  /**
   * Number of columns a grid cell should span.
   * @default 1
   */
  colSpan?: number | ((params: GridCellParams) => number | undefined);
  …
}
```

> ⚠ When using `colSpan`, some other features may be pointless or may not work as expected (depending on the data model). To avoid confusing grid layout, consider disabling the following features for the column(s) that are affected by `colSpan`:
>
> - [sorting](/components/data-grid/sorting/#disable-the-sorting)
> - [filtering](/components/data-grid/filtering/#disable-the-filters)
> - [column reorder](/components/data-grid/columns/#column-reorder)
> - [hiding columns](/components/data-grid/columns/#column-visibility)

### Number signature

> The `colSpan` number signature allows to span all the cells in the column:

Reviewer: ```suggestion
The number signature sets **all cells in the column** to span a given number of columns.
```

```ts
interface GridColDef {
  colSpan?: number;
}
```

{{"demo": "ColumnSpanningNumber.js", "bg": "inline"}}

### Function signature

> The `colSpan` function signature is useful for spanning only specific cells in the column. The function receives [`GridCellParams`](/api/data-grid/grid-cell-params/) as the first argument:

Reviewer: ```suggestion
The function signature allows spanning only **specific cells** in the column. The function receives [`GridCellParams`](/api/data-grid/grid-cell-params/) as the first argument:
```

> The function signature allows spanning only **specific cells** in the column. The function receives [`GridCellParams`](/api/data-grid/grid-cell-params/) as an argument:

Reviewer 2: It has only one argument.

```suggestion
The function signature allows spanning only **specific cells** in the column.
The function receives [`GridCellParams`](/api/data-grid/grid-cell-params/) as argument:
```
